content/blog/2022/2022-09-02-junit5.md

```diff
@@ -0,0 +1,601 @@
+```
+
+```java
+public class CatTest {  
+	@Test  
+	public void testBean() {  
+	    Cat cat = new Cat("Mimi");  
+	  
+	    assertThat(cat, hasProperty("name", equalTo("Mimi")));  
+	}
+}
+```
+
+### equalToIgnoringCase
+It checks if two strings match regardless of capital or small letters.
+
+```java
+public class CatTest {  
+	@Test  
+	public void testStringEquality() {  
+	    String catNameInCaps = "RACHEL";  
+	  
+	    assertThat(catNameInCaps, equalToIgnoringCase("rachel"));  
+	}
+}
+```
+
+### containsString
+It checks if a string is contained in another string.
+
+```java
+public class CatTest {  
+	@Test  
+	public void testStringContains() {  
+	    String catName = "Joey The Cute";  
+	  
+	    assertThat(catName, containsString("Cute"));  
+	}
+}
+```
+
+## Assumptions
+We can think of assumptions like preconditions that have to be met for the test to pass. Let's check some examples.
+
+Say we have a `GoldFish` class
+```java
+public class GoldFish {  
+    private String name;  
+	private int age;
+	// constructor and getters
+}
+```
+
+```java
+public class GoldFishTest {  
+  @Test  
+  public void testBooleanAssumption() {  
+        GoldFish goldFish = new GoldFish("Jelly", 1);  
+  
+	   assumeTrue(goldFish.getAge() > 0);  
+	   assertThat(goldFish.getName(), equalToIgnoringCase("Jelly"));  
+  }  
+}
+```
+We can also express it like
+```java
+@Test  
+public void testAssumption() {  
+    GoldFish goldFish = new GoldFish("Jelly", 1);  
+  
+    assumingThat(goldFish.getAge() > 0, 
+          () -> assertThat(goldFish.getName(), equalToIgnoringCase("Jelly")));  
+}
+```
+
+## Exception Testing
+As part of our test scenarios we might want to test that given certain conditions a particular exception is thrown. Let's check our JUnit5 helps us do this.
+
+JUnit5 offers us the `assertThrows` method, which takes the class of the exception we are excepting (or one of its super classes) as a first parameter. As for the second parameter it expects a Runnable that contains the code under test.
+
+Let's add a method to calculate the speed of GoldFish.
+
+```java
+public class GoldFish {  
+    private String name;  
+	private int age;
+	
+	public int calculateSpeed() {  
+	    if (age == 0){  
+	        throw new RuntimeException("This will fail :((");  
+	  }  
+	    return 10 / age;  
+	}
+	// constructor and getters
+}
+```
+Our test would look like this
+
+```java
+public class GoldFishTest {
+	@Test  
+	public void testException() {  
+	    GoldFish goldFish = new GoldFish("Goldy", 0);  
+	  
+	    RuntimeException exception = assertThrows(RuntimeException.class, goldFish::calculateSpeed);  
+	  
+	    assertThat(exception.getMessage(), equalToIgnoringCase("This will fail :(("));  
+	}
+}
+```
+
+## Parameterized Testing
+The idea behind parameterized testing is to execute the same test but with different parameters. JUnit5 offers a rich API to handle this, however, we are going to briefly explore the most common way to do it.
```

Comment:
If you tell the reader we're only looking at parts of it, at least leave a link to more info :)